### PR TITLE
Travis ci

### PR DESCRIPTION
**Pull request by andrewbird**
Let's create the PR, anything else can wait!

### CODE REVIEW

**Comment by andrewbird** [comment]
Hey tests failed I need to fix something!

**Comment by andrewbird** [comment]
Seem to be getting an fdpp build error
~~~
clang++ -std=c++11 -c -fno-threadsafe-statics -fno-rtti -fpic -iquote /home/travis/build/stsp/dosemu2/localfdpp.git/fdpp/../hdr -DI386 -DWITHFAT32 -I . -I /home/travis/build/stsp/dosemu2/localfdpp.git/fdpp/../kernel/ -I /home/travis/build/stsp/dosemu2/localfdpp.git/fdpp -DDATADIR=/home/travis/build/stsp/dosemu2/localfdpp/share/fdpp -DKRNL_NAME=fdppkrnl.sys -DKERNEL_VERSION="0.1beta7 [GIT: `git describe`]" -Wall -ggdb3 -O2 -Wno-format-invalid-specifier -o blockio.o blockio.cc
In file included from blockio.cc:32:
In file included from /home/travis/build/stsp/dosemu2/localfdpp.git/fdpp/../kernel/globals.h:65:
In file included from /home/travis/build/stsp/dosemu2/localfdpp.git/fdpp/../kernel/glob_fd.h:3:
./glob_inc.h:4:10: fatal error: 'glob_asmdefs.h' file not found
#include <glob_asmdefs.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
makefile:132: recipe for target 'blockio.o' failed
make[1]: *** [blockio.o] Error 1
make[1]: Leaving directory '/home/travis/build/stsp/dosemu2/localfdpp.git/fdpp'
makefile:13: recipe for target 'all' failed
make: *** [all] Error 2
~~~

**Comment by stsp** [comment]
Should now be fixed.
Good, even LP missed that and your tests caught.

**Comment by andrewbird** [comment]
Perhaps I should have returned an error after the fdpp build error, and aborted the test run?

**Comment by stsp** [comment]
Why fdpp build failure is not a "generic"
build failure? I don't suppose the tests will
run if dosemu2 itself does not build.

**Comment by andrewbird** [comment]
I think errors are not fatal in the 'script' (A.K.A. testing) phase so I moved the build to 'install' phase. Obviously I don't know if this would behave as I expect, because everything is fine since you fixed the FDPP,

File: test/test_dos.py [hunk 1]
```diff
@@ -303,6 +303,1073 @@ def test_systype(self):
...
+
+        # compile sources
+        mkcom(ename, r"""
+.text
+.code16
+
+    .globl  _start16
+_start16:
+
+    push    %%cs
```
Comment (stsp) [inline test/test_dos.py]: Embedding asm into python? Why?

File: .travis.yml [hunk 2]
```diff
@@ -1,43 +1,54 @@
...
-      - libao-dev
-      - libvdeplug-dev
-      - libreadline-dev
-      - binutils-dev
-      - pkg-config
-      - python-nose
-      - python-pexpect
-      - python-pkgconfig
-    update: true
+# Doesn't work in bionic beta
```
Comment (stsp) [inline .travis.yml]: Could you please squash that?

**Comment by stsp** [comment]
Looks cool!
Have you considered tests that are already
in `src/tests`? If not too difficult, would be nice
adding also them. Otherwise we have 2 dirs
with different tests, which is a bit confusing.

**Comment by andrewbird** [comment]
> Have you considered tests that are already in src/tests?

I'm not familiar with them, the build is currently failing, what do they test?
~~~
ajb@polly:/clients/common/dosemu2.git/src/tests$ make
gcc -m32 -Wall -O2 -g -fno-strict-aliasing  -o test-i386 test-i386.c test-i386-code16.S test-i386-vm86.S -lm
test-i386.c: In function ‘sig_handler’:
test-i386.c:1917:20: error: dereferencing pointer to incomplete type ‘struct ucontext’
            (long)uc->uc_mcontext.gregs[REG_TRAPNO],
                    ^~
test-i386.c: In function ‘sig_trap_handler’:
test-i386.c:2165:39: error: dereferencing pointer to incomplete type ‘struct ucontext’
     printf("EIP=" FMTLX "\n", (long)uc->uc_mcontext.gregs[REG_EIP]);
                                       ^~
Makefile:11: recipe for target 'test-i386' failed
make: *** [test-i386] Error 1
~~~

**Comment by stsp** [comment]
Fixed tests.
They check vm86.
Exceptions, trap flags and all that.
They do not test any DOS but dosemu itself.

**Comment by andrewbird** [comment]
Mmm, better but this might be a 32bit thing?
~~~
$ make
gcc -m32 -Wall -O2 -g -fno-strict-aliasing  -o test-i386 test-i386.c test-i386-code16.S test-i386-vm86.S -lm
test-i386.c: In function ‘test_code16’:
test-i386.c:1485:5: warning: asm operand 2 probably doesn’t match constraints
     asm volatile ("lcall %1, %2"
     ^~~
test-i386.c:1489:5: warning: asm operand 3 probably doesn’t match constraints
     asm volatile ("lcall %2, %3"
     ^~~
test-i386.c:1493:5: warning: asm operand 2 probably doesn’t match constraints
     asm volatile ("lcall %1, %2"
     ^~~
test-i386.c:1485:5: error: impossible constraint in ‘asm’
     asm volatile ("lcall %1, %2"
     ^~~
test-i386.c:1489:5: error: impossible constraint in ‘asm’
     asm volatile ("lcall %2, %3"
     ^~~
test-i386.c:1493:5: error: impossible constraint in ‘asm’
     asm volatile ("lcall %1, %2"
     ^~~
Makefile:11: recipe for target 'test-i386' failed
make: *** [test-i386] Error 1
~~~

**Comment by stsp** [comment]
Please try:
`gcc -m32 -Wall -O2 -g -fno-strict-aliasing  -o test-i386 test-i386.c test-i386-code16.S test-i386-vm86.S -lm -fno-pic`

**Comment by stsp** [comment]
Maybe even `-fno-pic -fno-pie` at the end.

**Comment by andrewbird** [comment]
That builds cleanly, but ends up with a segfault when run
~~~
arplw A=1234567b B=0762123d R=1234567b z=0
arplw A=12345679 B=0762123f R=1234567b z=1
arplw A=12345679 B=0762123d R=12345679 z=0
Segmentation fault (core dumped)
~~~

BTW this test needs splitting up to individual function tests.

**Comment by stsp** [comment]
Doesn't segfault for me.
But this doesn't matter.
The file of an interest is `test-i386.exe` not `test-i386`.

**Comment by andrewbird** [comment]
Okay, that's a djgpp compiled exe, that'll need to wait until I figure out what to do about that.

**Comment by stsp** [comment]
Indeed.
So why asms, autoexecs and all that are hardcoded in a python script?

**Comment by andrewbird** [comment]
@stsp There's something odd going on here. I added `valgrind` to the list of prerequisites as the FDPP plugin isn't built without it, but now seem to be getting a general exception at run time, see the travis log. Funny thing is I don't see it here on my 32bit machine.

**Comment by stsp** [comment]
That was a bug.
Please keep building fdpp as before.
As for the crash - can we ssh there and see ourselves?

**Comment by andrewbird** [comment]
Thanks, I've dropped the last commit and repushed, so it'll get retested now.

> As for the crash - can we ssh there and see ourselves?

I don't believe so.

**Comment by andrewbird** [comment]
Yep still failing
https://travis-ci.com/stsp/dosemu2/builds/120472105#L9919

**Comment by stsp** [comment]
If it fails w/o any patch, how can we bisect that?

**Comment by andrewbird** [comment]
Only way I can think of is to try checking out the first FDPP commit that has this API version and see if it works there. I just did a similar thing locally to find https://github.com/stsp/fdpp/issues/94 problem commit. It will be a little tedious remotely.

**Comment by andrewbird** [comment]
I found a good version, remote bisect progressing slowly...

**Comment by andrewbird** [comment]
Seems like the first bad commit is 
$ git bisect good
97f5a3253905a324c82961721bbe490421706ee6 is the first bad commit
commit 97f5a3253905a324c82961721bbe490421706ee6
Author: Stas Sergeev <stsp@users.sourceforge.net>
Date:   Tue Jul 23 22:38:01 2019 +0300

    intr.asm: rewrite some funcs in C [fixes #87]

 kernel/intr.asm | 38 +-------------------------------------
 kernel/proto.h  |  8 --------
 kernel/task.c   | 32 ++++++++++++++++++++++++++++++++
 3 files changed, 33 insertions(+), 45 deletions(-)

**Comment by stsp** [comment]
In #887 I checked that this now works.
You just need to rebase to devel and remove
the bisect commit.

**Comment by andrewbird** [comment]
Thanks, tests are all working again.

**Pull request closed without merging**